products/sle/main.pm

```diff
@@ -418,6 +419,47 @@ logcurrentenv(
       SLE_PRODUCT SPLITUSR VIDEOMODE)
 );
 
+sub parse_condition {
+    my ($schedule, $condition_name) = @_;
+    # Get condition hash
+    my $condition = $schedule->{conditional_schedule}->{$condition_name};
+    # Iterate over variables in the condition
+    foreach my $var (keys %{$condition}) {
+        next unless my $val = get_var($var);
+        # If value of the variable matched the conditions
+        # Iterate over the list of the modules to be loaded
+        for my $test (@{$condition->{$var}->{$val}}) {
+            loadtest $test;
+        }
+    }
+}
+
+sub parse_test_suite_setting {
+    my ($setting) = shift;
+    # allow comma spearated list
+    for my $test (split(',', get_var($setting))) {
+        loadtest $test;
+    }
+}
+
+if (my $yamlfile = get_var('YAML_SCHEDULE')) {
+    my $schedule = YAML::Tiny::LoadFile(dirname(__FILE__) . '/../../' . $yamlfile);
+    #schedule/lvm_thin_provisioning.yml
+    for my $module (@{$schedule->{schedule}}) {
+        if ($module =~ s/\{\{(.*)\}\}/$1/) {
```

Comment:
Just to keep the regexp sane, why not `=` or `< stuff >`? With `{{ stuff }}` I guess would be easier to simply miss one curly brace. (And as personal preference, I avoid quoting too much...)

Reply:
We have used `{{}}`, to be honest I guess I will replace it with & and * which is more widely used way of referencing things and that makes it more clear to which part we are referring to.

Reply:
Nice :) works for me